doc/source/train/user-guides.rst

```diff
@@ -18,4 +18,5 @@ Ray Train User Guides
     user-guides/fault-tolerance
     user-guides/monitor-your-application
     user-guides/reproducibility
+    user-guides/move-collate-to-data
```

Comment:
this should probably go into the training ingest section?

https://docs.ray.io/en/latest/train/user-guides/data-loading-preprocessing.html